packages/ocs/block-pool/body.tsx

```diff
@@ -79,6 +86,64 @@ export const BlockPoolBody = (props: BlockPoolBodyPros) => {
     []
   );
 
+  const [data, loaded, loadError] = useK8sList(CephBlockPoolModel);
+
+  const existingNames = React.useMemo(
+    () => (loaded && !loadError ? data?.map((data) => getName(data)) : []),
+    [data, loaded, loadError]
+  );
+
+  const fieldRequirements = [
+    t('No more than 253 characters'),
+    t('Starts and ends with a lowercase letter or number'),
+    t('Only lowercase letters, numbers, non-consecutive periods, or hyphens'),
+    t('Cannot be used before'),
+  ];
+
+  const schema = Yup.object({
+    newPoolName: Yup.string()
+      .required()
+      .max(253, fieldRequirements[0])
+      .matches(
+        validationRegEx.startAndEndsWithAlphanumerics,
+        fieldRequirements[1]
+      )
+      .matches(
+        validationRegEx.alphaNumericsPeriodsHyphensNonConsecutive,
+        fieldRequirements[2]
+      )
+      .test(
+        'unique-name',
+        fieldRequirements[3],
+        (value: string) => !existingNames.includes(value)
+      ),
+  });
+  const resolver = useYupValidationResolver(schema);
+  const {
+    control,
+    watch,
+    formState: { isValid },
+  } = useForm({
+    mode: 'all',
+    reValidateMode: 'onChange',
+    resolver: resolver,
```

Comment:
use currying with "yupValidationResolver" so to pass on this schema...
```suggestion
    resolver: yupValidationResolver(schema),
```

Reply:
will continue as hook like we agreed on call so  this change is unnecessary